examples/vae/main.py

```diff
@@ -142,30 +139,37 @@ def eval(model, eval_ds, z):
 
 
 def main(argv):
-  key = random.PRNGKey(0)
+  del argv
+  rng = random.PRNGKey(0)
+  rng, key = random.split(rng)
+
   train_ds = tfds.load('mnist', split=tfds.Split.TRAIN)
   train_ds = train_ds.cache().shuffle(1000).batch(FLAGS.batch_size)
   test_ds = tfds.as_numpy(tfds.load('mnist', split=tfds.Split.TEST, batch_size=-1))
 
-  _, params = VAE.init_by_shape(key, [((1, 784), jnp.float32)])
+  _, params = VAE.init_by_shape(
+      key, [((1, 784), jnp.float32)], z_rng=random.PRNGKey(0))
   vae = nn.Model(VAE, params)
 
   optimizer = optim.Adam(learning_rate=FLAGS.learning_rate).create(vae)
 
+  rng, z_key, eval_rng = random.split(rng, 3)
+  z = random.normal(z_key, (64, 20))
+
   for epoch in range(FLAGS.num_epochs):
     for batch in tfds.as_numpy(train_ds):
+      rng, key = random.split(rng)
       batch['image'] = batch['image'].reshape(-1, 784) / 255.0
```

Comment:
Why are images flattened and  then unflattened again in `eval`? If flattening serves a purpose, consider moving the first flattening op into the IO pipeline

Reply:
flattening is now part of the input pipeline